### ISSUE TITLE
Version `0.1.1.0` does not build on Windows.

### ISSUE THREAD

**Issue by jonathanknowles** [obsidiansystems/commutative-semigroups#17]
Hi there!

It seems that [version `0.1.1.0`](https://hackage.haskell.org/package/commutative-semigroups-0.1.1.0) of `commutative-semigroups` does not build on Windows machines.  😿 

I've tested with the following GHC major versions:
- `8.10`, `9.0`, `9.2`, `9.4`, `9.6`, `9.8`

In general, compilation fails with an error similar to:
```
Preprocessing library for commutative-semigroups-0.1.1.0..
Building library for commutative-semigroups-0.1.1.0..
[1 of 2] Compiling Numeric.Product.Commutative ( src\Numeric\Product\Commutative.hs, dist\build\Numeric\Product\Commutative.o )
[2 of 2] Compiling Data.Semigroup.Commutative ( src\Data\Semigroup\Commutative.hs, dist\build\Data\Semigroup\Commutative.o )

src\Data\Semigroup\Commutative.hs:28:1: error:
Error:     Could not find module ‘GHC.Event’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
28 | import GHC.Event (Event, Lifetime)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
See [example build log](https://github.com/jonathanknowles/quickcheck-monoid-subclasses/actions/runs/8590668657/job/23544464598#step:7:33).

Looking at the history, PR https://github.com/obsidiansystems/commutative-semigroups/pull/16 added instances of `Commutative` for the `Event` and `Lifetime` data types, exported from `GHC.Event`.

However, according to [`base.cabal`](https://gitlab.haskell.org/ghc/ghc/-/blob/master/libraries/base/base.cabal?ref_type=heads#L276-L297), the `GHC.Event` module is not available on Windows systems:
```cabal
    if os(windows)
        exposed-modules:
              GHC.IO.Encoding.CodePage.API
            , GHC.IO.Encoding.CodePage.Table
            , GHC.Conc.Windows
            , GHC.Conc.WinIO
            , GHC.Conc.POSIX
            , GHC.Conc.POSIX.Const
            , GHC.Windows
            , GHC.Event.Windows
            , GHC.Event.Windows.Clock
            , GHC.Event.Windows.ConsoleEvent
            , GHC.Event.Windows.FFI
            , GHC.Event.Windows.ManagedThreadPool
            , GHC.Event.Windows.Thread
            , GHC.IO.Handle.Windows
            , GHC.IO.Windows.Handle
            , GHC.IO.Windows.Encoding
            , GHC.IO.Windows.Paths
    else
        exposed-modules:
            GHC.Event
```

I'm guessing that one possible fix would be to make the import of `GHC.Event` (and the associated instances) conditional on the OS, and exclude it if the OS is Windows.

**Comment by Ericson2314** [comment]
Yes, that would indeed be the fix. If you could submit a PR that would be greatly appreciated!

**Comment by blamario** [comment]
Or we can just eliminate the two instances on all platforms. Sorry, I had no idea any `base` modules were platform-dependent.

**Comment by jonathanknowles** [comment]
@Ericson2314, @blamario 

Many thanks for looking at this issue! 🙌🏻 

I would be okay with either of the proposals:
- make the instances conditional on OS; or
- remove the instances altogether (as in #18).

Since there is already PR #18, I'm wondering if there's anything I can do to help get it merged and a new release made?

I also have a small proposal, which is to add Windows to the CI build matrix:
- b63b981487eaa19047f087b314dc5fd50d8b4af7 (Add Windows to CI build matrix)
(once it's possible to build on Windows again)

Many thanks again!

Jonathan

**Comment by blamario** [comment]
Thanks for chiming in @jonathanknowles. I don't need help with the PR, I was just waiting to see if there was any objection to removing the instances.

**Comment by blamario** [comment]
Let me know if you see any issue with the candidate release: https://hackage.haskell.org/package/commutative-semigroups-0.2/candidate

**Comment by endgame** [comment]
I'm wondering if there's also some metadata fix we should apply here. If those modules are only present on GNU/Linux (or non-Windows?) builds of GHC, then it would be good to have some way of indicating that to Cabal. Does a metadata revision allow checking for Windows and depending on `base < 0` if so? If now, should we consider marking the `0.1.1.0` release as broken?

**Comment by blamario** [comment]
I do intend to mark the `0.1.1.0` release as broken. Going meta, I think no platform-specific module should ever have been added to `base`.

**Comment by blamario** [comment]
I've published the `0.2` candidate and deprecated `0.1.1.0`

**Comment by jonathanknowles** [comment]
@blamario Many thanks!

I've created a PR that adds Windows to CI here: https://github.com/obsidiansystems/commutative-semigroups/pull/20
